plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationPuckManager.kt

```diff
@@ -187,6 +191,24 @@ internal class LocationPuckManager(
       *targets,
       options = options
     )
+    updateMaxPulsingRadiusToFollowAccuracyRing(*radius)
+  }
+
+  /**
+   * Update maximum radius for pulsing puck to follow location accuracy radius.
+   * since pulsing radius is in pixels and location accuracy is in meters, we convert meters to pixel using
+   * projection delegate.
+   */
+  @VisibleForTesting(otherwise = PRIVATE)
+  internal fun updateMaxPulsingRadiusToFollowAccuracyRing(vararg radius: Double) {
+    if (settings.pulsingMaxRadius.toInt() == LocationComponentConstants.PULSING_MAX_RADIUS_FOLLOW_ACCURACY.toInt()) {
+      val metersPerPixelAtLocation =
+        delegateProvider.mapProjectionDelegate.getMetersPerPixelAtLatitude(
+          delegateProvider.mapCameraManagerDelegate.cameraState.center.latitude(),
+          delegateProvider.mapCameraManagerDelegate.cameraState.zoom
+        )
+      animationManager.updatePulsingRadius(radius.last() / metersPerPixelAtLocation, settings)
```

Comment:
is it possible, that `metersPerPixelAtLocation` can be `0`?

Reply:
I guess not. will ask gl-native as well.

Reply:
Confirming with gl-native, this function will not return 0.0. there are safeguards on both scale factor (zoom) and max_latitude, so we can leave it the same.